slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -81,6 +83,24 @@ void MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
     }
 }
 
+std::string MapBuilder::SaveMapToStream() {
+    std::string buf;
+    const std::string filename = "temp-SaveState.pbstream";
+    cartographer::io::ProtoStreamWriter writer(filename);
+    map_builder_->SerializeState(/*include_unfinished_submaps=*/false, &writer);
```

Comment:
[q] Does `map_builder_mutex` need to be locked here? Or does `SerializeState` cover it?

Reply:
added!